Fix compare view overlap on short viewports

arottersman opened the pull request:
### Overview

The compare view charts overlapped with the modeling info below them on short viewports:
![image](https://cloud.githubusercontent.com/assets/7633670/25629834/f94cc54e-2f38-11e7-8d45-974a29d276d2.png)

If we move the modifications table down, it will be entirely off the screen on shorter viewports, so this PR just sets the entire compare column to scroll.

It also makes some styling updates per @jfrankl's suggestion: the spacing around the modification was unnecessarily large. I've also centered the "Navigate Scenarios" text on the back arrow.

Connects #179 

### Demo
![hv2qavh0sr](https://cloud.githubusercontent.com/assets/7633670/25629804/dd5eb996-2f38-11e7-9db9-5024eb046b81.gif)

### Testing instructions
- Pull and `./scripts/bundle.sh` (make sure updated css files get written)
- Draw an AoI over somewhere with data (ie the triangle near `yuba city`)
- Go to "Compare"
- Confirm view is useable at a variety of window sizes

rajadain commented:
Taking a quick look in IE and FF as well

rajadain commented:
FF works well. On IE11 in Windows, while there is no more overlap, there are no scrollbars either, making the content inaccessible:

![image](https://cloud.githubusercontent.com/assets/1430060/25636560/c0735d16-2f4f-11e7-84fd-5cc07b61fc76.png)

rajadain commented:
Scrollbars _are_ visible on FF on Windows

![image](https://cloud.githubusercontent.com/assets/1430060/25636965/1d9c39b2-2f51-11e7-89fc-a3fe9f4dfaa4.png)

Perhaps a little unseemly, but functional.

rajadain commented:
Also on Chrome on Windows

![image](https://cloud.githubusercontent.com/assets/1430060/25637251/5c22506c-2f52-11e7-8614-ba7e0113efce.png)

arottersman commented:
Fixup:
- I was calculating the chart+table height with a nested `calc()` (`$compare-height` mapped to another `calc`). IE11 couldn't figure it out, so I removed the intermediary variable
- Added `overflow-x: hidden` to the chart-table to remove the unseemly double scrollbar 
<img width="719" alt="screen shot 2017-05-03 at 9 38 27 am" src="https://cloud.githubusercontent.com/assets/7633670/25662985/7352e912-2fe4-11e7-962c-af77683177fa.png">


P.S. Where did you find olives??

rajadain commented:
> Where did you find the olives??

Just hadn't reprovisioned my VM in a while. They're gone now.

IE11 now has correct sidebars, and they work well. Unseemly as the horizontal sidebars may have been, they did serve a purpose. When the screen is too narrow, they help make hidden content accessible, like in this case:

![image](https://cloud.githubusercontent.com/assets/1430060/25676424/f7ac9cdc-300f-11e7-8408-bce7492c0113.png)

I think rather than setting `overflow-x: hidden`, we should figure out a better width for those columns. However, since this is likely to involve some experimentation, I'd suggest making another card for it so it can be pointed and accomplished on its own.

rajadain commented on src/icp/sass/pages/_compare.scss:
```diff
@@ -105,15 +110,20 @@
+          .chart-table-regions {
+              height: $compare-chart-table-height;
+              overflow-y: auto;
+              overflow-x: hidden;
```
Remove this line.

mmcfarland commented:
> I think rather than setting overflow-x: hidden, we should figure out a better width for those columns. 

A deep reach back to https://github.com/project-icp/bee-pollinator-app/pull/113 reveals some earlier discussion where we confirmed with the client that the compare chart is optimized for the likely max of 4 crops, and beyond that the individual crop dropdown will be used.  We shouldn't need to optimize the width for compare any further.

arottersman commented:
Thanks for your thorough testing, @rajadain! Squashed fixups. Going to merge. Jeff mentioned off hand that it's a little ridiculous to have the barchart bars so wide. If we ever address that, it may help with blueberries getting cut off in your screenshot above.
